var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -1281,3 +1320,46 @@ def test_mk(self):
             if os.access(f, os.R_OK):
                 return FileList(f)
         return FileList(find(self.prefix, "test.mk", recursive=True))
+
+    # See also find_system_libraries in lib/spack/llnl/util/filesystem.py
+    # where the similar list of paths is used.
+    sys_lib_paths = [
+        "/lib64",
+        "/lib",
+        "/usr/lib64",
+        "/usr/lib",
+        "/usr/local/lib64",
+        "/usr/local/lib",
+        "/usr/lib/x86_64-linux-gnu",
+    ]
+
+    def is_sys_lib_path(self, dir):
+        return dir in self.sys_lib_paths
+
+    @property
+    def xlinker(self):
+        return "-Wl," if "~cuda" in self.spec else "-Xcompiler="
+
+    # Similar to spec[pkg].libs.ld_flags but prepends rpath flags too.
+    # Also does not add system library paths as defined by 'sys_lib_paths'
+    # above -- this is done to avoid issues like this:
+    # https://github.com/mfem/mfem/issues/1088.
+    def ld_flags_from_library_list(self, libs_list):
+        flags = [
+            "%s-rpath,%s" % (self.xlinker, dir)
+            for dir in libs_list.directories
+            if not self.is_sys_lib_path(dir)
+        ]
+        flags += ["-L%s" % dir for dir in libs_list.directories if not self.is_sys_lib_path(dir)]
```

Comment:
Could you use `spack.util.environment.filter_system_paths`?

Reply:
We probably could. I remember that someone explicitly added the path `/usr/lib/x86_64-linux-gnu` to `mfem`'s `sys_lib_paths` (I suspect because they needed it) -- it's one path that is not included in the list used by `filter_system_paths`, so we'll probably want to filter one more time if we use `filter_system_paths`.

Reply:
> I remember that someone explicitly added the path /usr/lib/x86_64-linux-gnu to mfem's sys_lib_paths

I missed that - we should provide a convenience method for that. Until then I don't think it makes sense to ask for that refactor (including in this PR).